packages/discovery-react-components/src/components/DocumentPreview/components/PdfHighlight/utils/textBoxMapping/getTextBoxMapping.ts

```diff
@@ -30,28 +29,29 @@ export function getTextBoxMappings<
   const source = new Source(sourceLayout, targetLayout);
   const builder = new TextBoxMappingBuilder();
 
-  target.processText((targetCellId, targetText, markTargetAsMapped) => {
-    const matchInSource = source.findMatch(targetCellId, targetText);
-    if (matchInSource) {
-      const mappedTargetCells = markTargetAsMapped(matchInSource.matchLength);
-
-      let mappedSourceFullSpan: TextSpan = [0, 0];
-      mappedTargetCells.forEach(targetCell => {
-        const mappedSourceSpan = matchInSource.markSourceAsMapped(targetCell.text);
-        if (mappedSourceSpan) {
-          builder.addMapping(
-            { cell: matchInSource.cell, span: mappedSourceSpan },
-            { cell: targetCell }
-          );
-          mappedSourceFullSpan = spanMerge(mappedSourceFullSpan, mappedSourceSpan);
-        }
-      });
-      if (spanLen(mappedSourceFullSpan) > 0) {
-        matchInSource.markSourceMappedBySpan(mappedSourceFullSpan);
+  for (const minMatchLength of [27, 9, 3, 1]) {
+    debug('getTextBoxMapping: processText with minMatchLength: %d', minMatchLength);
```

Comment:
Match longer chunks first to fix scattered highlights
<img width="2501" alt="image" src="https://user-images.githubusercontent.com/19485344/156345744-d8ec2e19-73dc-4da6-879e-61f655a8eaf6.png">